documentation/concepts.md

```diff
@@ -3,17 +3,115 @@ layout: page
+```mermaid
+graph LR;
+    A("internal<br>(zone)") -->|policyA| B("external<br>(zone)")
+    B -->|policyB| A
+```
 
-firewalld does not depend on NetworkManager, but the use is recommended. If NetworkManager is not used, there are some limitations: firewalld will not get notified about network device renames. If firewalld gets started after the network is already up, the connections and manually created interfaces are not bound to a zone. You can add them to a zone with `firewall-cmd [--permanent] --zone=zone --add-interface=interface`, but make sure that if there's a `/etc/sysconfig/network-scripts/ifcfg-<interface>`, the zone specified there with `ZONE=zone` is the same (or both are empty/missing for default zone), otherwise the behaviour would be undefined.
+Policies may apply firewall rules to all traffic types. This includes
+filtering input, forwarded, and output traffic; as well as performing
+network address translations (NAT).
 
-firewalld provides support for [zones](zone/), [services](service/), [IPSets](ipset/) and [ICMP types](icmptype/).
+## Zone Principles
 
-There is also a so called [direct interface](direct-interface.html) for use in daemons and applications and also to be able to add firewall rules, that are not supported yet in firewalld directly.
+Firewalld follows some strict principles in regards to zones.
+
+1. traffic ingresses one and only one zone
+2. traffic egresses one and only one zone
+3. a zone defines a level of trust
+4. intra-zone (within the same zone) is allowed by default
+5. inter-zone (zone to zone) is denied by default
+
+Principle 4 and 5 are actually a consequence of Principle 3.
+
+Principle 4 is configurable via zone option `--remove-forward`.
+Principle 5 is configurable by adding new policies.
+
+## Firewall Rules
+
+Policies apply firewall rules in a stateful, unidirectional manner. This
+means you only need to consider one direction of the traffic. The
+traffic's return path is implicitly allowed due to firewalld's stateful
+filtering.
+
+Policies are associated with an ingress zone and an egress zone. The
+ingress zone is where the traffic originated (received). The egress zone
+is where the traffic leaves (sent).
+
+In this example, `myPolicy` will filter forwarded traffic originating
+from the `internal` ingress zone and destined for the `external` egress
+zone.
+
+```mermaid
+graph LR;
+    A("internal<br>(zone)") -->|myPolicy| B("external<br>(zone)")
+```
+
+This is a common scenario for a home network. To allow the LAN
+(internal) access the internet you can allow all traffic by setting the
+default target of the policy using `--set-target`.
+
+```
+# firewall-cmd --policy myPolicy --set-target ACCEPT
+```
+
+Alternatively, a strict network may allow only a subset of traffic. In
+this example, only `http` and `https` are allowed; other traffic will be
+rejected.
+
+```
+# firewall-cmd --policy myPolicy --add-service http
```

Comment:
I think this is where users get lost... there is no indication in this command that any zones are being affected. Does this command 'work' because the firewalld default zone on the machine on the strict network is 'internal'?

Reply:
No. This command is affecting a policy, not a zone. Note the `--policy` option.

Reply:
So then what causes `myPolicy` to be applied to any zones in particular? I seem to have missed that.

Reply:
In this example `myPolicy` affects traffic forwarded the `internal` zone to the `external` zone (explained in lines 51-53). It does not "apply to a zone" directly.

https://github.com/firewalld/firewalld.github.io/pull/25/commits/f68ad2461f121e561564a131dc05f0ff8a93d629#diff-296328860e666fb622185f2b4951481a189a5f689bbbea8cc2d7eabd0d50a057R51-R53

Reply:
OK, that's helpful. I guess my remaining concern is learning how that configuration occurred... should I assume that some command *before* the one shown created `myPolicy` and connected it to those zones? If so, then I'm happy :-)

Reply:
Yes. I didn't show the creation of `myPolicy` because I was trying to show only the concepts. Would it help if I showed the CLI output for `myPolicy`? It's below.

```
[root@vm-bos-rhel8 ~]# firewall-cmd --info-policy myPolicy
myPolicy (active)
  priority: -1
  target: CONTINUE
  ingress-zones: internal
  egress-zones: external
  services: 
  ports: 
  protocols: 
  masquerade: no
  forward-ports: 
  source-ports: 
  icmp-blocks: 
  rich rules: 
```

And these are the commands that create `myPolicy`:
```
# firewall-cmd --permanent --new-policy myPolicy
# firewall-cmd --permanent --policy myPolicy --add-ingress-zone internal
# firewall-cmd --permanent --policy myPolicy --add-egress-zone external
```

Reply:
In my opinion that would help quite a lot :-)

Reply:
I pushed the update. Please take a look.